Assets/Script/StateScripts/GameOverManager.cs

```diff
@@ -0,0 +1,61 @@
+﻿using UnityEngine;
+using UnityEngine.SceneManagement;
+
+public class GameOverManager : MonoBehaviour
+    {
+        public GameObject player; // Tham chieu den doi tuong la nguoi choi
+        public float screenBottomY; //Y-coordinate cua day man hinh
+
+        void Start()
+        {
+            screenBottomY = Camera.main.ViewportToWorldPoint(new Vector3(0, 0, 0)).y;
+        }
+
+        void Update()
+        {
+            //Kiem tra xem ke dich co cham vao day man hinh khong
+            CheckEnemiesFalling();
+        }
+
+        private void OnTriggerEnter2D(Collider2D other)
+        {
+            // Goi ham kiem tra neu ke dich va cham 
+            CheckEnemyCollision(other);
+        }
+
+        void CheckEnemyCollision(Collider2D other)
```

Comment:
why this function is needed instead of combining these conditions into  "OnTriggerEnter2D(Collider2D other)" function ?

Reply:
OnTriggerEnter2D is a special method in Unity and it is called automatically when another collider collides with the collider of the object it is attached to. If you rename it to CheckEnemyCollision, Unity will not recognize it as a collision method and will not call it automatically.